File: app/src/main/java/com/appsbygu/kadai_weather_java/interfaces/IWeatherService.java
package com.appsbygu.kadai_weather_java.interfaces;

import com.appsbygu.kadai_weather_java.models.apis.Cities.Rss;
import com.appsbygu.kadai_weather_java.models.apis.Weathers.Weather;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface IWeatherService {
    @GET("forecast/rss/primary_area.xml")
    Call<Rss> listCitis();

    @GET("forecast/webservice/json/v1")
    Call<Weather> weatherInfo(@Query("city") String city_code);
}

File: app/src/main/java/com/appsbygu/kadai_weather_java/interfaces/IWeatherService.kt
package com.appsbygu.kadai_weather_java.interfaces

import com.appsbygu.kadai_weather_java.models.apis.cities.Rss
import com.appsbygu.kadai_weather_java.models.apis.weathers.Weather

import retrofit2.http.GET
import retrofit2.http.Query

interface IWeatherService {
    @GET("forecast/rss/primary_area.xml")
    fun listCitis(): io.reactivex.Observable<Rss>

    @GET("forecast/webservice/json/v1")
    fun weatherInfo(@Query("city") city_code: String): io.reactivex.Observable<Weather>
}
